lib/publiccloud/ecr.pm

```diff
@@ -0,0 +1,44 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper class for Amazon Elastic Container Registry (ECR)
+#
+# Maintainer: Ivan Lausuch <ilausuch@suse.de>, qa-c team <qa-c@suse.de>
+# Documentation: https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html
+
+package publiccloud::ecr;
+use Mojo::Base 'publiccloud::aws';
+use testapi;
+
+sub push_container_img {
+    my ($self, $name) = @_;
+
+    my $aws_account_id = $self->get_user_id();
+    my $region         = get_var('PUBLIC_CLOUD_REGION',                'eu-central-1');
+    my $repository     = get_var("PUBLIC_CLOUD_CONTAINER_IMAGES_REPO", 'suse-qec-testing');
```

Comment:
I would try to use the openqa variables in the test and pass here only parameters.
```suggestion
    my ($self, $name, %args) = @_;
    $args{region} //= $self->region();
    $args{repo} //= get_var("PUBLIC_CLOUD_CONTAINER_IMAGES_REPO", 'suse-qec-testing');

    my $aws_account_id = $self->get_user_id();
```

Reply:
done